cmd/init.go

```diff
@@ -9,15 +9,47 @@ import (
 	"github.com/crytic/medusa/compilation"
 	"github.com/crytic/medusa/fuzzing/config"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 )
 
+// Get supported platforms for customized static completions of "init" flag `$ medusa init <tab> <tab>`
+// and to cache supported platforms for CLI arguments validation
+var supportedPlatforms = compilation.GetSupportedCompilationPlatforms()
+
 // initCmd represents the command provider for init
 var initCmd = &cobra.Command{
 	Use:   "init [platform]",
 	Short: "Initializes a project configuration",
 	Long:  `Initializes a project configuration`,
 	Args:  cmdValidateInitArgs,
 	RunE:  cmdRunInit,
+	// Run dynamic completion of nouns
+	ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		// Gather a list of flags that are available to be used in the current command but have not been used yet
+		var unusedFlags []string
+
+		// Examine all the flags, and add any flags that have not been set in the current command line
+		// to a list of unused flags
+		cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+			if !flag.Changed {
+				// When adding a flag to a command, include the "--" prefix to indicate that it is a flag
+				// and not a positional argument. Additionally, when the user presses the TAB key twice after typing
+				// a flag name, the "--" prefix will appear again, indicating that more flags are available and that
+				// none of the arguments are positional.
+				unusedFlags = append(unusedFlags, "--"+flag.Name)
+			}
+		})
+
+		// If a default platform is not specified, add a list of available platforms to the list of unused flags.
+		// If the --target or --out flag is set, it is assumed that the default platform is being used.
+		if len(args) == 0 && !(cmd.Flag("target").Changed || cmd.Flag("out").Changed) {
+			unusedFlags = append(unusedFlags, supportedPlatforms...)
+		}
```

Comment:
So although I think this is a great idea I feel like it might be a bit confusing to the user? especially b/c specifying a platform is optional with `init`. Idk I'm fine with keeping it in but when I was testing it it felt a bit off.

@Xenomega what do you think?

Reply:
Or maybe we keep it and potentially generalize `!(cmd.Flag("target").Changed || cmd.Flag("out").Changed)`. Is it possible to make it such that "if any flag is used, then don't add supportedPlatforms to the list".

Reply:
Maybe when we `VisitAll` we can maintain a boolean that says `flagUsed`. If `flag.Changed` is true for any flag, that `flagUsed` can be `true` and then we can check `if len(args) == 0 && !flagUsed { // add supportedPlatforms }`

Reply:
[906a4ac](https://github.com/crytic/medusa/pull/79/commits/906a4aca25259f4bb03c93a23e14a216e7e5d272)